src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        public Node(T item) {
+            this.item = item;
+        }
+    }
+
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if ((index > size) || (index < 0)) {
+            throw new IndexOutOfBoundsException();
+        }
+        Node<T> newNode = new Node<>(value);
+        if (first == null) {
+            first = last = newNode;
+        } else if (index == 0) {
+            newNode.next = first;
+            first.prev = newNode;
+            first = newNode;
+        } else if (index == size) {
+            last.next = newNode;
+            newNode.prev = last;
+            last = newNode;
+        } else {
+            Node<T> previous = getNodeByIndex(index - 1);
+            newNode.next = previous.next;
+            previous.next.prev = newNode;
+            previous.next = newNode;
+            newNode.prev = previous;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T node : list) {
+            add(node);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return getNodeByIndex(index).item;
```

Comment:
Wouldn't it then be better to call the checkIndex method in the getNodeByIndex method?

Reply:
I've tried but  getNodeByIndex method is called in add (T value, int index) method and there is another logic of checkIndex method (size not size - 1)